### PR TITLE
Update prometheus-operator deps

### PR DESCRIPTION
**Pull request by LiliC**
Pinning to latest commit of prometheus-operator and updating the deps along the way as we updated them upstream as well.

cc @brancz @s-urbaniak @paulfantom

### CODE REVIEW

**Comment by LiliC** [comment]
/test e2e-aws-operator

Do not see anything failed 🤔

File: go.mod [hunk 1]
```diff
@@ -5,28 +5,31 @@ go 1.12
 require (
 	github.com/Jeffail/gabs v1.1.1
 	github.com/ant31/crd-validation v0.0.0-20180801212718-38f6a293f140 // indirect
-	github.com/coreos/prometheus-operator v0.0.0-00010101000000-000000000000
-	github.com/emicklei/go-restful v2.8.0+incompatible // indirect
+	github.com/coreos/prometheus-operator v0.32.1-0.20191022141338-971190798bb6
```
Comment (paulfantom) [inline go.mod]: v0.32? But prometheus-operator is already at v0.33 :man_shrugging:

Comment (LiliC) [reply go.mod]: No clue, this is what go modules does when I do `go get github.com/coreos/prometheus-operator@971190798bb6bafcd6c109dfa344be4458c39018`

vendored code is correct, I checked.

Comment (LiliC) [reply go.mod]: If I remove this, and run go get, then go modules complains and says that there is no version `v00000000000`

**Comment by LiliC** [comment]
> level=error msg="Cluster operator authentication Degraded is True with RouteHealthDegradedFailedGet: RouteHealthDegraded: failed to GET route: dial tcp: lookup oauth-openshift.apps.ci-op-bz45xrqw-2249a.origin-ci-int-aws.dev.rhcloud.com on 172.30.0.10:53: no such host"

🤔don't think that is related to my change? 

/retest

**Pull request merged**